parsl/serialize/errors.py

```diff
@@ -0,0 +1,35 @@
+from parsl.errors import ParslError
+
+
+class DeserializationError(ParslError):
+    """Failure at the deserialization of results/exceptions from remote workers.
+    """
+
+    def __init__(self, reason: str) -> None:
+        self.reason = reason
+
+    def __str__(self) -> str:
+        return f"Failed to deserialize objects. Reason: {self.reason}"
+
+
+class SerializationError(ParslError):
+    """Failure to serialize task objects.
+    """
+
+    def __init__(self, fname: str) -> None:
+        self.fname = fname
+        self.troubleshooting = "https://parsl.readthedocs.io/en/latest/faq.html#addressing-serializationerror"
+
+    def __str__(self) -> str:
+        return f"Failed to serialize objects for an invocation of function {self.fname}. Refer {self.troubleshooting}"
+
```

Comment:
I like this pointer to the documentation.  A couple of questions:
- Is there a way to test that this link is valid?  As in, during CI?  (And I _don't_ mean reaching out to rtd.io in every run.)
- I'm wondering if we might do something similar for the deserialization side.  I'm not seeing a way currently because not all deserialization errors manifest so cleanly or directly, of course.

Reply:
this is actually imported from a different PR #2839 (so this will disappear from this current PR if/when that is merged), and *that* PR inherits it from the original import location... 

That link doesn't even try to do any kind of versioning seriousness, so as soon as that stops being frequently asked, the link will break for the entire installed userbase... I can imagine various over-complicated uses of metaclasses or subclass hooks or etc that are URL+documentation aware... but that seems like a lot of work for an abandoned project.

These two errors are awkwardly asymmetrical in what they report, in part from a half hearted attempt previously to include reference links in some error messages but not others.